src/main/java/com/education/service/TimetableServiceImpl.java

```diff
@@ -92,48 +91,63 @@ public void update(long id, Timetable newTimetable) {
 
     /*
      * 1-read timetable base on id -->timetable:timetable
-     * 7-read all of timetables of this student --> timetablesByStudentId:List<Timetable>
-     * 8-loop in the timetablesByStudentId --> timetables
-     *  8-1- if the date of timetable equals of date of timetables
-     *    8-1-1 if Start timetable is before end of timetables and start of timetable is  after start of timetables or
+     * 2-read all of timetables of this student --> timetablesByStudentId:List<Timetable>
+     * 3-loop in the timetablesByStudentId --> timetables
+     *  3-1- if the date of timetable equals of date of timetables
+     *    3-1-1 if Start timetable is before end of timetables and start of timetable is  after start of timetables or
      *          end of timetable is before end of timetables and end of timetable is after start of timetables
-     *      8-1-1-1 throws an exception
-     *    8-1-2 else if start of timetable is equal of start of timetables
-     *      8-1-2-1 throws an exception END IF
+     *      3-1-1-1 throws an exception
+     *    3-1-2 else if start of timetable is equal of start of timetables
+     *      3-1-2-1 throws an exception END IF
      *
-     * 5-if the size of students in timetable is bigger than capacity of timetable
-     *  5-1- throws an exception
-     * 2-read all students of timetable -->allStudents:List<Students>
-     * 3-read a student by getById -->student: Student
-     * 4-add studentId to this List
-     * 6-set student to the timetable
+     * 4-if the size of students in timetable is bigger than capacity of timetable
+     *  4-1- throws an exception
+     * 5-read all students of timetable -->allStudents:List<Students>
+     * 6-read a student by getById -->student: Student
+     * 7-add studentId to this List
+     * 8-set student to the timetable
      * 9-save this timetable
      */
     @Override
     public void addStudentToTimetable(long timetableId, long studentId) {
-        Timetable timetable = getById(timetableId);
+        Timetable target = getById(timetableId);
+        validateStartAndEndOfTimetableWithOtherTimetables(studentId, target);
+        validateTargetTimetableCapacity(target);
```

Comment:
"Target" can be removed. Do not use variable names in your method. You may use this method later for another timetable which is not a target timetable.